corehq/apps/accounting/invoicing.py

```diff
@@ -754,8 +866,17 @@ def sms_billables_queryset(self):
             domain__in=self.subscribed_domains,
             is_valid=True,
             date_sent__gte=self.invoice.date_start,
-            date_sent__lt=self.invoice.date_end + datetime.timedelta(days=1),
-        ).order_by('-date_sent')
+
+            # Don't count any SMS billables towards the monthly total &
+            # limit if they were sent AFTER the end date of the current
+            # subscription related to this line item. Save it for the
+            # next subscription in the invoice.
+            date_sent__lt=(
+                self._end_date_count_sms if self._end_date_count_sms
+                else self.invoice.date_end + datetime.timedelta(days=1)
+            ),
+
+        ).order_by('date_sent')
```

Comment:
Its a bit uncommon to see a large comment in the middle of a function call. This could have also been written as:

```python
queryset = SmsBillable.objects.filter(
    domain__in=self.subscribed_domains,
    is_valid=True,
    date_sent__gte=self.invoice.date_start
).order_by('date_sent')

# comment
if self._end_date_count_sms:
    return queryset.filter(date_set__lt=self._end_date_count_sms)

return queryset.filter(date_set__lt=self.invoice.date_end + datetime.timedelta(days=1))
```

I've also seen it as constructing a dictionary of filters and then doing something like `return queryset.filter(**filters)`